File: 121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        Left = 0
        max = 0

        for Right in range(len(prices)):
            if prices[Right] < prices[Left]:
                Left = Right
            
            if  prices[Right] - prices[Left] > max :
                max =  prices[Right] - prices[Left]
            print(max)
        return max

File: 121.java
class Solution {
    public int maxProfit(int[] prices) {
        int mx = 0;

        int l = 0;
        int r = 1;

        while (r < prices.length){
            if(prices[r] - prices[l] > mx){
                mx = prices[r] - prices[l];
            }

            if (prices[r] < prices[l]){
                l = r;
            }
            r ++;

        }
        return mx;
        
    }
}